formbuilder/components/XLSDownloader.js

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import json2xls from "json2xls";
+import btoa from "btoa";
+
+export default function XLSDownloader(props) {
+  const filename = props.schema.title + ".xls";
+  const fields = props.fields;
+  const fieldNames = props.fields.map((key) => {
+    return props.schema.properties[key].title;
+  });
+
+  // Pre-processing of data
+  // json2xls takes json in the following format
+  // json_data = [
+  //   {
+  //     col1: row1value1,
+  //     col2: row1value2
+  //   },
+  //   {
+  //     col1: row2value1,
+  //     col2: row2value2
+  //   }
+  // ]
+  // After the pre-processing is done, we will have data in
+  // above mentioned format in `formatted_data` which can be
+  // then easily fed to `json2xls`.
+  var formatted_data = [];
+  var rows = props.records;
```

Comment:
Isn't this the simplest way to do what we want to do? This way it will be easily customizable in future. I personally prefer simplicity in code. What do you think?

Even if we use `map` function over the rows (list of objects), we will still need one `for` function inside to iterate over the different column names (`fieldNames`).